app/classes/token.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class Token
+  attr_accessor :hash
+
+  def initialize(hash = {})
+    @hash = hash
+  end
+
+  def encrypt_to_header
+    token_json = JSON.generate(@hash)
+    Token.crypt.encrypt_and_sign(token_json)
+  end
+
+  def self.decrypt_from_header(auth_header)
+    token = Token.crypt.decrypt_and_verify(auth_header)
+    hash = JSON.parse(token)
+    new(hash)
+  rescue ActiveSupport::MessageVerifier::InvalidSignature
+    nil
+  end
+
+  def self.crypt
+    @crypt ||= ActiveSupport::MessageEncryptor.new(
```

Comment:
I guessed at random about whether to use @ (class instance var) or @@ (class variable).  I suspect I was wrong, and this should be @@.  Joe, do you remember?  I looked it up for a while, and the thing that resonated with me is that class instance variables (@) are in a sense an "accident" of how ruby creates classes.   But class variables respect inheritance and behave how most people would expect class variables to act.  So I think think @@ is recommended over @...  I think...

Reply:
I think the relevant difference is that class variables can be modified by subclasses, which in many circumstances can lead to unpleasant surprises.  Good examples under "What is the difference between class variables and class instance variables?" at https://www.ruby-lang.org/en/documentation/faq/8/

Reply:
Ok so i'm going back to `@crypt` here then?